test/OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.test.cpp

```diff
@@ -0,0 +1,422 @@
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Coordinate/Position.hpp>
+#include <OpenSpaceToolkit/Physics/Coordinate/Velocity.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/LocalOrbitalFrameFactory.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinateSubset/CartesianPosition.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinateSubset/CartesianVelocity.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.hpp>
+
+#include <Global.test.hpp>
+
+using ostk::core::container::Array;
+using ostk::core::type::Real;
+using ostk::core::type::Size;
+using ostk::core::type::String;
+
+using ostk::mathematics::object::MatrixXd;
+using ostk::mathematics::object::Vector3d;
+using ostk::mathematics::object::VectorXd;
+
+using ostk::physics::coordinate::Frame;
+using ostk::physics::coordinate::Position;
+using ostk::physics::coordinate::Velocity;
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
+
+using ostk::astrodynamics::solver::FiniteDifferenceSolver;
+using ostk::astrodynamics::solver::LeastSquaresSolver;
+using ostk::astrodynamics::trajectory::LocalOrbitalFrameFactory;
+using ostk::astrodynamics::trajectory::State;
+using ostk::astrodynamics::trajectory::state::CoordinateSubset;
+using ostk::astrodynamics::trajectory::state::coordinatesubset::CartesianPosition;
+using ostk::astrodynamics::trajectory::state::coordinatesubset::CartesianVelocity;
+using ostk::astrodynamics::trajectory::StateBuilder;
+
+class OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver : public ::testing::Test
+{
+   protected:
+    void SetUp() override
+    {
+        // Define initial state (position and velocity)
+        const Position initialPosition = Position::Meters({1.0, 0.0, 0.0}, Frame::GCRF());
+        const Velocity initialVelocity = Velocity::MetersPerSecond({0.0, 0.0, 0.0}, Frame::GCRF());
+        trueState_ = {Instant::J2000(), initialPosition, initialVelocity};
+
+        // Define observations (true states with noise)
+        const Size observationCount = 100;
+        const Real timeStep = 0.1;  // seconds
+
+        for (Size i = 0; i < observationCount; ++i)
+        {
+            const Real time = timeStep * (Real)i;
+            const Real position = std::cos(time);
+            const Real velocity = -std::sin(time);
+
+            const Position truePosition = Position::Meters({position, 0.0, 0.0}, Frame::GCRF());
+            const Velocity trueVelocity = Velocity::MetersPerSecond({velocity, 0.0, 0.0}, Frame::GCRF());
+
+            // Add noise to the true state
+            std::mt19937 rng(42);  // Fixed seed for reproducibility
+            std::uniform_real_distribution<double> dist(-0.01, 0.01);
```

Comment:
This is the wrong distribution to use, it should be a normal distribution, that way you can exactly match it with the sigmas